presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveCoercionPolicy.java

```diff
@@ -57,7 +57,7 @@ private boolean canCoerce(HiveType fromHiveType, HiveType toHiveType)
             return toHiveType.equals(HIVE_BYTE) || toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG);
         }
         if (toType instanceof VarcharType) {
-            return fromHiveType.equals(HIVE_BYTE) || fromHiveType.equals(HIVE_SHORT) || fromHiveType.equals(HIVE_INT) || fromHiveType.equals(HIVE_LONG);
+            return fromHiveType.equals(HIVE_BYTE) || fromHiveType.equals(HIVE_SHORT) || fromHiveType.equals(HIVE_INT) || fromHiveType.equals(HIVE_LONG) || fromHiveType.equals(HIVE_FLOAT) || fromHiveType.equals(HIVE_DOUBLE) || fromType instanceof DecimalType;
```

Comment:
Looks like unfortunate code duplication with conversion rule in `presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java`

we should have a conversions factory singleton and use it in both places.

Reply:
Could you elaborate on what you mean by this?

Do you mean the `HivePageSource#createCoercer()` should be replaced with a call to a Factory that returns Coercers? similar to the `DecimalCoercers` class at https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/coercions/DecimalCoercers.java?

Reply:
let's not do this in this PR, but the current approach is simply errorprone.
Even more so, that the conditions here are expressed differently than those in the other file.

cc @losipiuk